client/client.go

```diff
@@ -132,12 +138,13 @@ func New(ctx context.Context, address string, opts ...ClientOpt) (*Client, error
 	}
 	if authority == "" {
 		// authority as hostname from target address
-		uri, err := url.Parse(address)
-		if err != nil {
-			return nil, err
-		}
 		authority = uri.Host
 	}
+	if uri.Scheme == "tcp" {
+		// remove tcp scheme from address, since default dialer doesn't expect that
```

Comment:
Could you add an inline link to https://github.com/grpc/grpc/blob/master/doc/naming.md, to help the future reader :tada:

Sorry for leaving this PR unreviewed for so long, the way that gRPC handles dialing/addresses is really bizarre to get my head around.

Reply:
included the naming as a comment 👍